File: sdk/node/src/examples/create_project.ts
import Project from "../project";
import Service from "../service";
import path from "path";
import { connect } from "../client";

function main() {
  const deno = new Service()
    .withName("deno-fresh")
    .withCommand("./dev.ts")
    .withEnv({
      PORT: "8000",
    });

  const projectDir = path.resolve("../../examples/deno-fresh");

  connect()
    .newProject()
    .withName("deno-example")
    .withContext(projectDir)
    .withService(deno)
    .stdout();
}

main();

File: sdk/rust/examples/create_project.rs
use superviseur_client::{client::connect, service::new_service};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let deno_fresh = new_service()
        .with_name("deno-fresh")
        .with_command("./dev.ts")
        .with_env("PORT", "8000");

    connect()
        .new_project("deno-example")
        .with_context("/Users/tsirysandratraina/Documents/GitHub/superviseur/examples/deno-fresh")
        .with_service(deno_fresh)
        .stdout()
        .await?;

    Ok(())
}
